scripts/DeployAgTokenSideChainMultiBridge.s.sol

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "forge-std/Script.sol";
+import "./utils/Constants.s.sol";
+import "utils/src/CommonUtils.sol";
+import { AgTokenSideChainMultiBridge } from "contracts/agToken/AgTokenSideChainMultiBridge.sol";
+import { LayerZeroBridgeToken } from "contracts/agToken/layerZero/LayerZeroBridgeToken.sol";
+import { ICoreBorrow } from "contracts/interfaces/ICoreBorrow.sol";
+import { Treasury } from "contracts/treasury/Treasury.sol";
+import { ImmutableCreate2Factory } from "contracts/interfaces/external/create2/ImmutableCreate2Factory.sol";
+
+contract DeployAgTokenSideChainMultiBridge is Script, CommonUtils {
+    using stdJson for string;
+
+    function run() external {
+        /** TODO  complete */
+        string memory stableName = vm.envString("STABLE_NAME");
+        address expectedAddress = vm.envAddress("EXPECTED_ADDRESS");
+        uint256 totalLimit = vm.envUint("TOTAL_LIMIT");
+        uint256 hourlyLimit = vm.envUint("HOURLY_LIMIT");
+        uint256 chainTotalHourlyLimit = vm.envUint("CHAIN_TOTAL_HOURLY_LIMIT");
+        bool mock = vm.envOr("MOCK", false);
+        /** END  complete */
+
+        uint256 deployerPrivateKey = vm.deriveKey(vm.envString("MNEMONIC_MAINNET"), "m/44'/60'/0'/0/", 0);
+        address deployer = vm.addr(deployerPrivateKey);
+        string memory jsonVanity = vm.readFile(JSON_VANITY_PATH);
+        bytes32 salt = bytes32(abi.encodePacked(deployer, abi.encodePacked(uint96(jsonVanity.readUint("$.init")))));
+        uint256 chainId = vm.envUint("CHAIN_ID");
+
+        string memory json = vm.readFile(JSON_ADDRESSES_PATH);
+        address proxyAdmin;
+        address coreBorrow;
+        if (vm.keyExistsJson(json, ".proxyAdmin")) {
+            proxyAdmin = vm.parseJsonAddress(json, ".proxyAdmin");
+        } else {
+            proxyAdmin = _chainToContract(chainId, ContractType.ProxyAdmin);
+        }
+        if (vm.keyExistsJson(json, ".coreBorrow")) {
```

Comment:
is the json per chain or something like this?

Reply:
the json so far is for all chains. In my mind we deploy one chain at a time so no real need for multiples